src/server/game/Battlefield/Battlefield.cpp

```diff
@@ -814,6 +812,7 @@ Creature* Battlefield::SpawnCreature(uint32 entry, float x, float y, float z, fl
 
     // Set creature in world
     map->AddToMap(creature);
+    creature->setActive(true);
```

Comment:
Not sure about it.
[valgrind (2).log](https://github.com/azerothcore/azerothcore-wotlk/files/6450460/valgrind.2.log)
There is a lot of memleaks related to that function, but not sure what is the cause. This is only my suspicion.